Clinical-Genomics/scout
# Cloud custom igv tracks from Amazon S3 storage

Allows to display igv tracks from private Amazon buckets.

Any type of track with any name can be used, the important is to describe it properly in the config file (see later in this text). Users can choose which cloud tracks they want to visualize in igv, or they can decide to use only the default tracks, disregarding the cloud ones.

![image](https://user-images.githubusercontent.com/28093618/83254620-62f82600-a1af-11ea-8d00-4b3d2e82a059.png)


**Preparing for testing**:
1. Edit the scout/server/config.py file and add these lines:
```
CUSTOM_IGV_TRACKS = {
    "cosmic_coding_v90" : {
        "37":{
            "description" : "Cosmic coding mutations v90 hg19",
            "file_name": "CosmicCodingMuts_v90_hg19.vcf.gz",
            "type": "variant",
            "format": "vcf",
            "displayMode": "squished",
            "genome_build": "37",
            "access_type": "credentials",
            "index_format": "tbi"
        },
        "38":
        {
            "description" : "Cosmic coding mutations v90 hg38",
            "file_name": "CosmicCodingMuts_v90_hg38.vcf.gz",
            "type": "variant",
            "format": "vcf",
            "displayMode": "squished",
            "genome_build": "37",
            "access_type": "credentials",
            "index_format": "tbi"
        }
    },
    "cosmic_non_coding_v90": {
        "37":{
            "description" : "Cosmic non coding variants v90 hg19",
            "file_name": "CosmicNonCodingVariants_v90_hg19.vcf.gz",
            "type": "variant",
            "format": "vcf",
            "displayMode": "squished",
            "genome_build": "37",
            "access_type": "credentials",
            "index_format": "tbi"
        },
        "38":
        {
            "description" : "Cosmic non coding variants v90 hg38",
            "file_name": "CosmicNonCodingVariants_v90_hg38.vcf.gz",
            "type": "variant",
            "format": "vcf",
            "displayMode": "squished",
            "genome_build": "37",
            "access_type": "credentials",
            "index_format": "tbi"
        }
    }
}
```

1. Edit the scout/server/config.py file and add also these cloud settings (ask @northwestwitch for the real params):
```
# Cloud storage resources accession params
REGION_NAME = "eu-north-1"
ACCESS_KEY = "cloud_access_key"
SECRET_ACCESS_KEY = "password_access_key"
BUCKET_NAME = "name_of_bucket"
BUCKET_FOLDER = "folder_on_the_cloud" # Optional
```

**How to test**:
1. From a local instance of scout go to a MT variant (because it has alignment file)
1. Make sure that there is a select box displaying available custom tracks
1. Test viewing the alignment with and without one or more custom tracks

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
